ExadelTimeTrackingSystem/ExadelTimeTrackingSystem.WebAPI/Controllers/TasksController.cs

```diff
@@ -97,8 +102,35 @@ public async Task<ActionResult<List<BulkCreateTaskDTO>>> BulkCreateAsync([FromBo
         {
             var cancellationToken = CancellationTokenCreator.Create(_options.CurrentValue.TimeOutSeconds);
             cancellationToken.ThrowIfCancellationRequested();
-            var tasksDto = await _service.BulkCreateAsync(bulkCreateTaskDto, cancellationToken);
+            var tasksDto = await _taskService.BulkCreateAsync(bulkCreateTaskDto, cancellationToken);
             return Created(string.Empty, tasksDto);
         }
+
+        [HttpPost("emailtest")]
+        public async Task<ActionResult> SendEmailToApproversAsync([FromQuery] Guid employeeId)
+        {
+            var cancellationToken = CancellationTokenCreator.Create(_options.CurrentValue.TimeOutSeconds);
+            cancellationToken.ThrowIfCancellationRequested();
+
+            List<string> approverNames = new List<string>();
+            List<string> approverEmails = new List<string>();
+            var employeeName = await _userService.GetNameAsync(employeeId, cancellationToken);
+            var approverIdList = await _taskService.GetApproversAsync(employeeId, cancellationToken);
+
+            foreach (var approver in approverIdList)
+            {
+                approverNames.Add(await _userService.GetNameAsync(approver, cancellationToken));
+                approverEmails.Add(await _userService.GetEmailAsync(approver, cancellationToken));
```

Comment:
you are calling database twice. That is not good. Retrieve Email and Name with one call. You can create custom projection class for it